pkg/workflows/wasm/runner_wasip1.go

```diff
@@ -24,103 +25,121 @@ func log(respptr unsafe.Pointer, respptrlen int32)
 //go:wasmimport env fetch
 func fetch(respptr unsafe.Pointer, resplenptr unsafe.Pointer, reqptr unsafe.Pointer, reqptrlen int32) int32
 
-const uint32Size = int32(4)
-
-func bufferToPointerLen(buf []byte) (unsafe.Pointer, int32) {
-	return unsafe.Pointer(&buf[0]), int32(len(buf))
-}
+//go:wasmimport env emit
+func emit(respptr unsafe.Pointer, resplenptr unsafe.Pointer, reqptr unsafe.Pointer, reqptrlen int32) int32
 
 func NewRunner() *Runner {
 	l := logger.NewWithSync(&wasmWriteSyncer{})
 
 	return &Runner{
-		sendResponse: func(response *wasmpb.Response) {
-			pb, err := proto.Marshal(response)
-			if err != nil {
-				// We somehow couldn't marshal the response, so let's
-				// exit with a special error code letting the host know
-				// what happened.
-				os.Exit(CodeInvalidResponse)
-			}
+		sendResponse: sendResponseFn,
+		sdkFactory: func(sdkConfig *RuntimeConfig, id string, md *capabilities.RequestMetadata, opts ...func(*RuntimeConfig)) *Runtime {
+			sdkConfig.RequestID = &id
+			sdkConfig.Metadata = md
```

Comment:
:/ This is worse IMO since you're coupling id + RequestMetadata with RuntimeConfig through a function. They are all being passed in from the same place, why not move these lines so that they happen before passing it in?